src/usage/search.md

```diff
@@ -1,83 +1,136 @@
-* The `WHERE` clause in a query is just simple like `(id % 2 = 0) AND (age >  50)`.
+* Search without filter and transaction
+* Returning insufficient vectors is acceptable
+* The `vbase` search mode fails due to out-of-memory
 
-Prefilter is also used in internal implementation for handling deleted rows in `pgvecto.rs`.
+### `vbase`
 
-Prefilter may have a negative impact on precision. Test the precision before using it.
+`vbase` is the recommended search mode when any filter is enabled.
 
-Prefilter is enabled by default because it almost only works if you create a multicolumn vector index.
+In this mode, the filter is applied after a batch of vectors is returned.
+It will continuously request the next batch until the collected vectors are sufficient.
 
-## Options
+In most cases, `vbase` mode would return enough vectors for your filter.
+For how it works, see the thesis [VBASE: Unifying Online Vector Similarity Search and Relational Queries via Relaxed Monotonicity](https://www.usenix.org/conference/osdi23/presentation/zhang-qianxi).
 
-Search options are specified by PostgreSQL GUC. You can use `SET` command to apply these options in session or `SET LOCAL` command to apply these options in transaction.
+It's recommended in these situations:
 
-Runtime parameters for planning a query:
+* Search with filter or transaction
+* Returning sufficient vectors is important
+* Tired of tuning `vectors.hnsw_ef_search` in `basic` mode
 
-| Option               | Type    | Range              | Default   | Description                                                                  |
-| -------------------- | ------- | ------------------ | --------- | ---------------------------------------------------------------------------- |
-| vectors.enable_index | boolean |                    | `on`      | Enables or disables the query planner's use of vector index-scan plan types. |
-| vectors.search_mode  | enum    | `"basic", "vbase"` | `"basic"` | Search mode.                                                                 |
+You can enable `vbase` by `SET vectors.search_mode = vbase;`.
+
+## Prefilter
+
+`prefilter` is an enhancement strategy that can be set in any search mode.
+It is enabled by default.
+
+If enabled, an additional filter is applied to select vectors before they are collected.
+This reduces expensive distance calculations, and all collected vectors will match the filter.
+
+The acceleration ratio is positively correlated to the filtering rate.
+A filtering rate of 10% will result in a 10x acceleration of distance calculations.
+
+::: details
+Example: 10% filtering rate
+```sql
+SELECT * FROM generate_series(1, 10) WHERE generate_series <= 1;
+```
+:::
+
+Prefilter may have a negative impact on precision.
+If you need a high level of precision, please test your scenarios and consider turning it off:
+```sql 
+ALTER SYSTEM SET vectors.enable_prefilter=off;
```

Comment:
I do no think someone needs to turn it off. Any option will influence precision.